new project template: "main" field in package.json

With a "main" field that would point at "lib/index.js", users would be free to change this path after the fact, while at the same time being able to run `npm run develop`. That is, `bin/mage` would be able to `require(process.cwd())` instead of `process.cwd() + '/lib'`.

Are you sure this works? I tried quickly and got an error like this:

```shell
Error: Cannot find module '/Users/mt/Sources/mage-parser/test/fake-project'
```

`package.json` contains `"main": "./lib/index.js",`

--- a/bin/mage
+++ b/bin/mage
@@ -1,8 +1,4 @@
 #!/usr/bin/env node
 
-var path = require('path');
-var cwd = process.cwd();
-var libPath = path.join(cwd, 'lib');
-var app = require(libPath);
-
+var app = require(process.cwd());
 app.boot();

--- a/scripts/templates/javascript-project/package.json
+++ b/scripts/templates/javascript-project/package.json
@@ -5,6 +5,7 @@
   "repository": "%APP_REPO%",
   "author": "%APP_AUTHOR%",
   "license": "%APP_LICENSE%",
+  "main": "./lib/index.js",
   "scripts": {
     "help": "     echo 'Display help information (this screen)' &&   npm run",
     "mage": "     echo 'Run the mage binary'  &&                     mage",

--- a/scripts/templates/typescript-project/package.json
+++ b/scripts/templates/typescript-project/package.json
@@ -5,6 +5,7 @@
   "repository": "%APP_REPO%",
   "author": "%APP_AUTHOR%",
   "license": "%APP_LICENSE%",
+  "main": "./lib/index.ts",
   "scripts": {
     "help": "     echo 'Display help information (this screen)' &&   npm run",
     "mage": "     echo 'Run the mage binary'  &&                     ts-mage",

